app/schemas/tech.relaycorp.courier.data.database.AppDatabase/1.json

```diff
@@ -2,24 +2,42 @@
   "formatVersion": 1,
   "database": {
     "version": 1,
-    "identityHash": "4056a2f52fe6f834146b1ab85d35e7a8",
+    "identityHash": "011b4f5a90e3c27ac4a0d9daf6061ec3",
     "entities": [
       {
-        "tableName": "cargo",
-        "createSql": "CREATE TABLE IF NOT EXISTS `${TABLE_NAME}` (`recipientAddress` TEXT NOT NULL, `messageId` TEXT NOT NULL, `creationTime` INTEGER NOT NULL, `expirationTime` INTEGER NOT NULL, `storagePath` TEXT NOT NULL, `size` INTEGER NOT NULL, `senderAddress` TEXT NOT NULL, PRIMARY KEY(`recipientAddress`, `messageId`))",
+        "tableName": "Message",
+        "createSql": "CREATE TABLE IF NOT EXISTS `${TABLE_NAME}` (`recipientAddress` TEXT NOT NULL, `recipientType` TEXT NOT NULL, `senderAddress` TEXT NOT NULL, `messageId` TEXT NOT NULL, `messageType` TEXT NOT NULL, `creationTime` INTEGER NOT NULL, `expirationTime` INTEGER NOT NULL, `storagePath` TEXT NOT NULL, `size` INTEGER NOT NULL, PRIMARY KEY(`senderAddress`, `messageId`))",
         "fields": [
           {
             "fieldPath": "recipientAddress",
             "columnName": "recipientAddress",
             "affinity": "TEXT",
             "notNull": true
           },
+          {
+            "fieldPath": "recipientType",
+            "columnName": "recipientType",
+            "affinity": "TEXT",
+            "notNull": true
+          },
```

Comment:
Can you specify constraints here? If so, I'd suggest to mimic an enum field with something like `CHECK( recipientType IN ('private','public') )`

Reply:
I had to confirm wether the Android 5 sqlite version supported `CHECK`. It's a version from 2014, but it does support it. But our ORM, Room, does not support `CHECK` constraints out of the box. We would need to create table ourselves and disable auto-creation in Room.

What is happening right now, is that we're enforcing that constraint on the Type we're using. Since we're always getting and setting data through those types, Room is enforcing that constraint. As long as we always use Room api to interact with the database, the constraint is kept. What do you think?

Reply:
Sounds good! I thought it'd be easier, but given that I agree keeping it as it is is the best course of action.